### PR TITLE
JAMES-3945 advertize that subaddressing is supported

### PR DESCRIPTION
**Pull request by florentos17**
[jira JAMES-3945 ticket for subaddressing](https://issues.apache.org/jira/browse/JAMES-3945)

subaddressing has now been implemented on the backend side, but not fully deployed. we need to have the frontend automatically detect if the backend supports subaddressing or not.

thus i modified the `urn:apache:james:params:jmap:mail:shares` capability to add a `subaddressingSupported` property (boolean). if this property is not defined, then the frontend should assume it is not supported.

### CODE REVIEW

**Comment by chibenwa** [comment]
Question 1: Why do we need this?

Question 2: Why isn't the doc updated? ( https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621/doc/specs/spec/mail/rights.mdown#defined-capability )

Question 3: Why ISSUE-1324? THis is NOT an ASF bugtracker. Can we git inspect to find a related issue where we could explain what we do here? And use it as a PR title / commit message prefix?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/Capability.scala [hunk 1]
```diff
@@ -317,7 +317,7 @@ case object DelegationCapabilityFactory extends CapabilityFactory {
 }
 
 final case class SharesCapabilityProperties() extends CapabilityProperties {
-  override def jsonify(): JsObject = Json.obj()
+  override def jsonify(): JsObject = Json.obj("subaddressingSupported" -> true)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/Capability.scala:320]: No impact on tests

Comment (florentos17) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/Capability.scala:320]: i just wrote one, i am checking that everything is fine and i push

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/Capability.scala:320]: It should impact existing session integration tests, no need to write new test IMO

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**